crates/holochain/src/conductor/conductor.rs

```diff
@@ -869,11 +869,57 @@ mod network_impls {
                         let all_peers = peer_store.get_all().await?;
                         out.extend(all_peers);
                     }
-                    Ok(out)
+                    Ok(out.into_iter().collect())
                 }
             }
         }
 
+        /// Get signed agent info from the conductor for a given app
+        pub async fn get_app_agent_infos(
+            &self,
+            installed_app_id: &InstalledAppId,
+            dna_hash: Option<DnaHash>,
+        ) -> ConductorApiResult<Vec<Arc<AgentInfoSigned>>> {
+            // Get app info to know which DNAs belong to this app
+            let app_info = self.get_app_info(installed_app_id).await?.ok_or_else(|| {
+                ConductorApiError::other(format!("App not installed: {}", installed_app_id))
+            })?;
+
+            // Get all agent infos
+            let mut all_infos = self.get_agent_infos(None).await?;
```

Comment:
The code under this is well enough optimized that I don't think this is a huge cost but this is still allocating a list of agent infos that we only need some of. On a conductor with many apps, this is retrieving a much bigger list than we need.

I'd prefer flipping this on its head and getting the DNAs from the `AppInfo` first. Then requesting agent infos per DNA and merging those lists into a set.

I see that's not possible right now with the `get_agent_infos` taking a `CellId` but as we just discussed on call, that doesn't make much sense. I think once that's changed to take a DNA hash, this code can be more selective about what it retrieves.